src/cache/lru-cache.ts

```diff
@@ -6,24 +6,25 @@
  * The cache is implemented as a Map, which is a built-in JavaScript object.
  * The Map object holds key-value pairs and remembers the order of key-value pairs as they were inserted.
  */
-export class LRUCache implements Map<string, string> {
-  private readonly cache = new Map<string, string>();
-  [Symbol.toStringTag]: string;
+export class LRUCache implements Set<string> {
+  private readonly cache = new Set<string>();
```

Comment:
👌 Nice that we are good to go with `Set` insertion order as well:
> You can iterate through the elements of a set in insertion order. The insertion order corresponds to the order in which each element was inserted into the set by the `add()`

([doc link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set))

Reply:
We will want to update the comment above this method explaining that though 😅